provider/aws/config.go

```diff
@@ -39,47 +39,63 @@ type AWSSessionConfig struct {
 	AssumeRoleExternalID string
 	APIRetries           int
 	Profile              string
+	DomainRolesMap       map[string]string
 }
 
-func CreateDefaultV2Config(cfg *externaldns.Config) awsv2.Config {
+type AWSZoneConfig struct {
+	Config         awsv2.Config
+	HostedZoneName string
+	Route53Config  Route53API
```

Comment:
On my phone so no lookup possible, so maybe I am wrong.
The name seems to be wrong. I think elsewhere we call it client iirc.